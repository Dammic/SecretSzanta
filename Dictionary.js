module.exports = {
    SocketEvents: {
        SelectName: 'CLIENT_SELECT_NAME',
        SyncLobby: 'SYNC_LOBBY',
        SyncPolicies: 'SYNC_POLICIES',
        IncreaseTrackerPosition: 'IncreaseTrackerPosition',
        PlayersListChanged: 'PLAYERS_LIST_CHANGED',
        SyncRoomsList: 'SYNC_ROOMS_LIST',
        RoomsListChanged: 'ROOMS_LIST_CHANGED',
        AllowEnteringRoom: 'ALLOW_ENTERING_ROOM',
        ServerWaitingForVeto: 'SERVER_WAITING_FOR_VETO',
        VetoVoteRegistered: 'VETO_VOTE_REGISTERED',
        CLIENT_SEND_MESSAGE: 'CLIENT_SEND_MESSAGE',
        CLIENT_JOIN_ROOM: 'CLIENT_JOIN_ROOM',
        CLIENT_LEAVE_ROOM: 'CLIENT_LEAVE_ROOM',
        CLIENT_GET_ROOM_DATA: 'CLIENT_GET_ROOM_DATA',
        CLIENT_CREATE_ROOM: 'CLIENT_CREATE_ROOM',
        CLIENT_ERROR: 'CLIENT_ERROR',
        VOTING_PHASE_START: 'VOTING_PHASE_START',
        CLIENT_VOTE: 'CLIENT_VOTE',
        VOTING_PHASE_NEWVOTE: 'VOTING_PHASE_NEWVOTE',
        VOTING_PHASE_REVEAL: 'VOTING_PHASE_REVEAL',
        START_GAME: 'START_GAME',
        CHANCELLOR_CHOICE_PHASE: 'CHANCELLOR_CHOICE_PHASE',
        BECOME_FACIST: 'BECOME_FACIST',
        KillSuperpowerUsed: 'KILL_SUPERPOWER_USED',
        PlayerKilled: 'PLAYER_KILLED',
        PlayerKicked: 'PLAYER_KICKED',
        PlayerBanned: 'PLAYER_BANNED',
        GameFinished: 'GAME_FINISHED',
        PresidentChoosePolicy: 'PRESIDENT_CHOOSE_POLICY',
        ChancellorChoosePolicy: 'CHANCELLOR_CHOOSE_POLICY',
        ChoosePolicy: 'CHOOSE_POLICY',
        NewPolicy: 'NEW_POLICY',
        ResetTracker: 'RESET_TRACKER',
        SetTimer: 'SET_TIMER',
    },
    PlayerDirection: {
        PLAYER_DIRECTION_LEFT: 'PLAYER_DIRECTION_LEFT',
        PLAYER_DIRECTION_UP: 'PLAYER_DIRECTION_UP',
        PLAYER_DIRECTION_RIGHT: 'PLAYER_DIRECTION_RIGHT',
    },
    PlayerRole: {
        ROLE_CHANCELLOR: 'ROLE_CHANCELLOR',
        ROLE_PRESIDENT: 'ROLE_PRESIDENT',
        ROLE_PREVIOUS_CHANCELLOR: 'ROLE_PREVIOUS_CHANCELLOR',
        ROLE_PREVIOUS_PRESIDENT: 'ROLE_PREVIOUS_PRESIDENT',
    },
    GamePhases: {
        GAME_PHASE_NEW: 'GAME_PHASE_NEW',
        START_GAME: 'START_GAME',
        GAME_PHASE_CHANCELLOR_CHOICE: 'GAME_PHASE_CHANCELLOR_CHOICE',
        GAME_PHASE_VOTING: 'GAME_PHASE_VOTING',
        GAME_PHASE_SUPERPOWER: 'GAME_PHASE_SUPERPOWER',
        GAME_PHASE_FINISHED: 'GAME_PHASE_FINISHED',
        PresidentPolicyChoice: 'GAME_PHASE_PRESIDENT_POLICY_CHOICE',
        ChancellorPolicyChoice: 'GAME_PHASE_CHANCELLOR_POLICY_CHOICE',
        ServerWaitingForVeto: 'GAME_PHASE_SERVER_WAITING_FOR_VETO',
        ServerAcceptedVeto: 'GAME_PHASE_SERVER_ACCEPTED_VETO',
        Paused: 'GAME_PHASE_PAUSE',
    },
    PlayerAffilications: {
        LIBERAL_AFFILIATION: 'LIBERAL_AFFILIATION',
        FACIST_AFFILIATION: 'FACIST_AFFILIATION',
        HITLER_AFFILIATION: 'HITLER_AFFILIATION',
    },
    MessagesTypes: {
        ERROR: 'ERROR',
        INFO: 'INFO',
    },
    ErrorMessages: {
        notOwner: 'This action is forbidden! You are not the owner!',
        notPresident: 'This action is reserved for president!',
        notChancellor: 'This action is reserved for chancellor!',
        NameTaken: 'The selected name is already taken',
    },
    ChoiceModeContexts: {
        ChancellorChoice: 'CONTEXT_CHANCELLOR_CHOICE',
        KillChoice: 'CONTEXT_KILL_CHOICE',
        KickChoice: 'CONTEXT_KICK_CHOICE',
        BanChoice: 'CONTEXT_BAN_CHOICE',
    },
    PolicyCards: {
        FacistPolicy: 'FACIST_POLICY',
        LiberalPolicy: 'LIBERAL_POLICY',
    },
    Views: {
        Home: 'HOME',
        Lobby: 'LOBBY',
        HowToPlay: 'HOW_TO_PLAY',
        News: 'NEWS',
        About: 'ABOUT',
        Game: 'GAME',
    },
    PlayerBoards: {
        SmallBoard: 'SMALL_BOARD',
        MediumBoard: 'MEDIUM_BOARD',
        LargeBoard: 'LARGE_BOARD',
    },
    GlobalRoomName: 'Global',
}
