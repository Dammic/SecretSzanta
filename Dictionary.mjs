export const SocketEvents = {
    SelectName: 'CLIENT_SELECT_NAME',
    SyncLobby: 'SYNC_LOBBY',
    SyncPolicies: 'SYNC_POLICIES',
    IncreaseTrackerPosition: 'IncreaseTrackerPosition',
    PlayersListChanged: 'PLAYERS_LIST_CHANGED',
    SyncRoomsList: 'SYNC_ROOMS_LIST',
    RoomsListChanged: 'ROOMS_LIST_CHANGED',
    AllowEnteringRoom: 'ALLOW_ENTERING_ROOM',
    ServerWaitingForVeto: 'SERVER_WAITING_FOR_VETO',
    VetoVoteRegistered: 'VETO_VOTE_REGISTERED',
    CLIENT_SEND_MESSAGE: 'CLIENT_SEND_MESSAGE',
    ClientGameNotification: 'CLIENT_GAME_NOTIFICATION',
    CLIENT_JOIN_ROOM: 'CLIENT_JOIN_ROOM',
    CLIENT_LEAVE_ROOM: 'CLIENT_LEAVE_ROOM',
    CLIENT_GET_ROOM_DATA: 'CLIENT_GET_ROOM_DATA',
    CLIENT_CREATE_ROOM: 'CLIENT_CREATE_ROOM',
    CLIENT_ERROR: 'CLIENT_ERROR',
    VOTING_PHASE_START: 'VOTING_PHASE_START',
    CLIENT_VOTE: 'CLIENT_VOTE',
    VOTING_PHASE_NEWVOTE: 'VOTING_PHASE_NEWVOTE',
    VOTING_PHASE_REVEAL: 'VOTING_PHASE_REVEAL',
    START_GAME: 'START_GAME',
    CHANCELLOR_CHOICE_PHASE: 'CHANCELLOR_CHOICE_PHASE',
    BECOME_FACIST: 'BECOME_FACIST',
    KillSuperpowerUsed: 'KILL_SUPERPOWER_USED',
    PlayerKilled: 'PLAYER_KILLED',
    PlayerKicked: 'PLAYER_KICKED',
    PlayerBanned: 'PLAYER_BANNED',
    GameFinished: 'GAME_FINISHED',
    PresidentChoosePolicy: 'PRESIDENT_CHOOSE_POLICY',
    ChancellorChoosePolicy: 'CHANCELLOR_CHOOSE_POLICY',
    ChoosePolicy: 'CHOOSE_POLICY',
    NewPolicy: 'NEW_POLICY',
    ResetTracker: 'RESET_TRACKER',
    SetTimer: 'SET_TIMER',
    DesignateNextPresident: 'DESIGNATE_NEXT_PRESIDENT',
    SuperpowerAffiliationPeekPlayerChoose: 'SUPERPOWER_AFFILIATION_PEEK_PLAYER_CHOOSE',
    SuperpowerAffiliationPeekAffiliationReveal: 'SUPERPOWER_AFFILIATION_PEEK_AFFILIATION_REVEAL',
    PeekCards: 'PEEK_CARDS',
    SetChooserPlayer: 'SET_CHOOSER_PLAYER',
}
export const PlayerDirection = {
    PLAYER_DIRECTION_LEFT: 'PLAYER_DIRECTION_LEFT',
    PLAYER_DIRECTION_UP: 'PLAYER_DIRECTION_UP',
    PLAYER_DIRECTION_RIGHT: 'PLAYER_DIRECTION_RIGHT',
}
export const PlayerRole = {
    ROLE_CHANCELLOR: 'ROLE_CHANCELLOR',
    ROLE_PRESIDENT: 'ROLE_PRESIDENT',
    ROLE_PREVIOUS_CHANCELLOR: 'ROLE_PREVIOUS_CHANCELLOR',
    ROLE_PREVIOUS_PRESIDENT: 'ROLE_PREVIOUS_PRESIDENT',
}
export const GamePhases = {
    GAME_PHASE_NEW: 'GAME_PHASE_NEW',
    START_GAME: 'START_GAME',
    GAME_PHASE_CHANCELLOR_CHOICE: 'GAME_PHASE_CHANCELLOR_CHOICE',
    GAME_PHASE_VOTING: 'GAME_PHASE_VOTING',
    GAME_PHASE_SUPERPOWER: 'GAME_PHASE_SUPERPOWER',
    GAME_PHASE_FINISHED: 'GAME_PHASE_FINISHED',
    PresidentPolicyChoice: 'GAME_PHASE_PRESIDENT_POLICY_CHOICE',
    ChancellorPolicyChoice: 'GAME_PHASE_CHANCELLOR_POLICY_CHOICE',
    ServerWaitingForVeto: 'GAME_PHASE_SERVER_WAITING_FOR_VETO',
    ServerAcceptedVeto: 'GAME_PHASE_SERVER_ACCEPTED_VETO',
    Paused: 'GAME_PHASE_PAUSE',
    DesignateNextPresidentPhase: 'GAME_PHASE_DESIGNATE_NEXT_PRESIDENT',
    PeekAffiliationSuperpowerPhase: 'PEEK_AFFILIATION_SUPERPOWER_PHASE',
    PeekCardsSuperpower: 'GAME_PHASE_PEEK_CARDS_SUPERPOWER',
    Ended: 'GAME_PHASE_ENDED',
}
export const PlayerAffilications = {
    LIBERAL_AFFILIATION: 'LIBERAL_AFFILIATION',
    FACIST_AFFILIATION: 'FACIST_AFFILIATION',
    HITLER_AFFILIATION: 'HITLER_AFFILIATION',
}
export const MessagesTypes = {
    ERROR: 'ERROR',
    INFO: 'INFO',
    STATUS: 'STATUS',
}
export const ErrorMessages = {
    notOwner: 'This action is forbidden! You are not the owner!',
    notPresident: 'This action is reserved for president!',
    notChancellor: 'This action is reserved for chancellor!',
    NameTaken: 'The selected name is already taken',
}
export const ErrorTypes = {
    DeniedRoomEntry: {
        BeganGame: 'DENIED_ROOM_ENTRY_BEGAN_GAME',
        FullRoom: 'DENIED_ROOM_ENTRY_FULL_ROOM',
        SamePlayerName: 'DENIED_ROOM_ENTRY_SAME_PLAYER_NAME',
    },
}
export const ChoiceModeContexts = {
    ChancellorChoice: 'CONTEXT_CHANCELLOR_CHOICE',
    KillChoice: 'CONTEXT_KILL_CHOICE',
    KickChoice: 'CONTEXT_KICK_CHOICE',
    BanChoice: 'CONTEXT_BAN_CHOICE',
    DesignateNextPresidentChoice: 'CONTEXT_DESIGNATE_NEXT_PRESIDENT_CHOICE',
    AffiliationPeekChoice: 'CONTEXT_AFFILIATION_PEEK_CHOICE',
}
export const PolicyCards = {
    FacistPolicy: 'FACIST_POLICY',
    LiberalPolicy: 'LIBERAL_POLICY',
}
export const Views = {
    Home: 'HOME',
    Lobby: 'LOBBY',
    HowToPlay: 'HOW_TO_PLAY',
    News: 'NEWS',
    About: 'ABOUT',
    Game: 'GAME',
}
export const PlayerBoards = {
    SmallBoard: 'SMALL_BOARD',
    MediumBoard: 'MEDIUM_BOARD',
    LargeBoard: 'LARGE_BOARD',
}
export const WinReasons = {
    fiveLiberalCards: '5 liberal policies have been enacted',
    hitlerDead: 'Hitler has been assassinated',
    sixFascistCards: '6 fascist policies have been enacted',
    hitlerBecameChancellor: 'Hitler has become chancellor after 4 fascist cards have been enacted',
}
export const GlobalRoomName = 'Global'

export const ErrorMappedMessages = {
    [ErrorTypes.DeniedRoomEntry.BeganGame]: 'The game has already began!',
    [ErrorTypes.DeniedRoomEntry.FullRoom]: 'The room is full!',
    [ErrorTypes.DeniedRoomEntry.SamePlayerName]: 'There is a player with the same name in the room!',
}
