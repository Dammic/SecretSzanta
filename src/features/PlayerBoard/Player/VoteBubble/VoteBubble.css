@import 'variables/variables.css';

.bubble {
    opacity: 0;
    position: absolute;
    z-index: 1;
    padding: 6px;

    width: var(--player-bubble-width);
    height: var(--player-bubble-height);
    
    max-width: 10vw;

    background-color: var(--player-bubble-color);
    transition: all 0.7s; 

    border: 1px solid var(--player-bubble-color);
    border-radius: var(--player-bubble-radius);
}

.bubbleLeft {
    vertical-align: center;
    left: var(--player-bubble-original-position);
    top: 0px;

    &:before {
        content: "";
        position: absolute;
        display: block;

        width: 0;
        height: 0;
        right: 95%;

        border-top: var(--player-bubble-triangle-size) solid transparent;
        border-bottom: var(--player-bubble-triangle-size) solid transparent;
        border-right: var(--player-bubble-triangle-size) solid var(--player-bubble-color);
    }

    &.active {
        opacity: 1;
        left: var(--player-bubble-activated-position);
    }
}

.bubbleRight {
    right: var(--player-bubble-original-position);
    top: 0px;

    &:after {
        content: "";
        position: absolute;
        display: block;

        width: 0;
        height: 0;
        vertical-align: center;
        left: 95%;

        border-top: var(--player-bubble-triangle-size) solid transparent;
        border-bottom: var(--player-bubble-triangle-size) solid transparent;
        border-left: var(--player-bubble-triangle-size) solid var(--player-bubble-color);
    }

    &.active {
        opacity: 1;
        right: var(--player-bubble-activated-position);
    }
}

.bubbleTop {
    top: var(--player-bubble-original-position);
    
    &:before {
        content: "";
        position: absolute;
        display: block;

        width: 0;
        height: 0;

        bottom: 95%;
        border-bottom: var(--player-bubble-triangle-size) solid var(--player-bubble-color);
        border-right: var(--player-bubble-triangle-size) solid transparent;
        border-left: var(--player-bubble-triangle-size) solid transparent;
    }

    &.active {
        opacity: 1;
        top: var(--player-bubble-activated-position);
    }
}

