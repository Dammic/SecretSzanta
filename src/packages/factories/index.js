export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList, aPlayer, aPlayersList } from './lobby'
