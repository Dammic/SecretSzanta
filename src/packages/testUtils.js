import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
