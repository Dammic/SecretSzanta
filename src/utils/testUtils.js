import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
