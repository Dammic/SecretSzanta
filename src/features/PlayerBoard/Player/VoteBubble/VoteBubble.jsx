import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { isUndefined, get } from 'lodash'
import { PlayerDirection, GamePhases } from '../../../../../Dictionary'

import styles from './VoteBubble.css'

export class VoteBubble extends React.PureComponent {
    static displayName = 'VoteBubble'
    static propTypes = {
        // parent
        playerName: PropTypes.string.isRequired,
        direction: PropTypes.string,

        // redux
        votes: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.bool, PropTypes.string])),
        gamePhase: PropTypes.string,
    }

    getVoteBubbleStyle = () => {
        const { direction } = this.props

        switch (direction) {
            case PlayerDirection.PLAYER_DIRECTION_LEFT:
                return classNames(styles.bubble, styles.bubbleLeft, { [styles.active]: this.isBubbleActive() })
            case PlayerDirection.PLAYER_DIRECTION_RIGHT:
                return classNames(styles.bubble, styles.bubbleRight, { [styles.active]: this.isBubbleActive() })
            default:
                return classNames(styles.bubble, styles.bubbleTop, { [styles.active]: this.isBubbleActive() })
        }
    }

    getVoteValue = () => {
        const { votes, playerName } = this.props
        const vote = get(votes, playerName)

        // it means the user has not voted yet, ex. he's not in the votes object
        if (isUndefined(vote)) {
            return undefined
        // it means the user has voted, but the votes have not been revealed yet
        } else if (vote === '') {
            return ''
        }
        return (vote ? 'JA' : 'NEIN')
    }

    isBubbleActive = () => {
        const { votes, gamePhase, playerName } = this.props
        return !isUndefined(get(votes, playerName)) && gamePhase === GamePhases.GAME_PHASE_VOTING
    }

    render() {
        const voteValue = this.getVoteValue()
        if (isUndefined(voteValue)) {
            return null
        }

        return (
            <div className={this.getVoteBubbleStyle()}>{voteValue}</div>
        )
    }
}

const mapStateToProps = ({ room }) => ({
    votes: room.votes,
    gamePhase: room.gamePhase,
})

export default connect(mapStateToProps)(VoteBubble)
