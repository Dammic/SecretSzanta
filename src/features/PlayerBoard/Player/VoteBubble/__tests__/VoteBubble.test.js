import React from 'react'
import { shallow } from 'enzyme'
import { VoteBubble } from '../VoteBubble'
import { expectMatchingSnapshot } from '/packages/testUtils'
import { PlayerDirection, GamePhases } from '../../../../../../Dictionary'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        playerName: 'mockPlayerName',
        direction: null,
        votes: { mockPlayerName: '' },
        gamePhase: null,
        ...propsOverrides,
    }
    const component = renderMethod(<VoteBubble {...props} />)
    return { props, component }
}

describe('<VoteBubble />', () => {
    it('renders active top bubble', () => {
        const { props } = setupProps({
            votes: { mockPlayerName: true },
            gamePhase: GamePhases.GAME_PHASE_VOTING,
        })

        expectMatchingSnapshot(<VoteBubble {...props} />)
    })
    it('does not render bubble at all if player has not voted (his voting value is undefined)', () => {
        const { props } = setupProps({
            votes: { mockPlayerName: undefined },
        })

        expectMatchingSnapshot(<VoteBubble {...props} />)
    })

    describe('isBubbleActive', () => {
        it('returns false if game phase is not defined and player has not voted', () => {
            const { component } = setupProps({
                votes: { mockPlayerName: undefined },
                gamePhase: null,
            })
            const instance = component.instance()
            expect(instance.isBubbleActive()).toEqual(false)
        })

        it('returns false if game phase is not voting and player has not voted', () => {
            const { component } = setupProps({
                votes: { mockPlayerName: undefined },
                gamePhase: 'someothergamephase',
            })
            const instance = component.instance()
            expect(instance.isBubbleActive()).toEqual(false)
        })

        it('returns false if game phase is voting and player has not voted', () => {
            const { component } = setupProps({
                votes: { mockPlayerName: undefined },
                gamePhase: GamePhases.GAME_PHASE_VOTING,
            })
            const instance = component.instance()
            expect(instance.isBubbleActive()).toEqual(false)
        })

        it('returns true if game phase is voting and player has voted (either true or false)', () => {
            const { component } = setupProps({
                votes: { mockPlayerName: true },
                gamePhase: GamePhases.GAME_PHASE_VOTING,
            })
            const instance = component.instance()
            expect(instance.isBubbleActive()).toEqual(true)

            component.setProps({ votes: { mockPlayerName: false } })
            expect(instance.isBubbleActive()).toEqual(true)
        })
    })

    describe('getVoteBubbleStyle', () => {
        describe('left', () => {
            it('renders inactive bubble', () => {
                const { component } = setupProps({
                    direction: PlayerDirection.PLAYER_DIRECTION_LEFT,
                })
                const instance = component.instance()
                instance.isBubbleActive = () => false
                const receivedClassName = instance.getVoteBubbleStyle()
                expect(receivedClassName).toEqual('bubble bubbleLeft')
            })

            it('renders active bubble', () => {
                const { component } = setupProps({
                    direction: PlayerDirection.PLAYER_DIRECTION_LEFT,
                })
                const instance = component.instance()
                instance.isBubbleActive = () => true
                const receivedClassName = instance.getVoteBubbleStyle()
                expect(receivedClassName).toEqual('bubble bubbleLeft active')
            })
        })

        describe('right', () => {
            it('renders inactive bubble', () => {
                const { component } = setupProps({
                    direction: PlayerDirection.PLAYER_DIRECTION_RIGHT,
                })
                const instance = component.instance()
                instance.isBubbleActive = () => false
                const receivedClassName = instance.getVoteBubbleStyle()
                expect(receivedClassName).toEqual('bubble bubbleRight')
            })

            it('renders active bubble', () => {
                const { component } = setupProps({
                    direction: PlayerDirection.PLAYER_DIRECTION_RIGHT,
                })
                const instance = component.instance()
                instance.isBubbleActive = () => true
                const receivedClassName = instance.getVoteBubbleStyle()
                expect(receivedClassName).toEqual('bubble bubbleRight active')
            })
        })

        describe('top', () => {
            it('renders inactive bubble', () => {
                const { component } = setupProps({
                    direction: PlayerDirection.PLAYER_DIRECTION_TOP,
                })
                const instance = component.instance()
                instance.isBubbleActive = () => false
                const receivedClassName = instance.getVoteBubbleStyle()
                expect(receivedClassName).toEqual('bubble bubbleTop')
            })

            it('renders active bubble', () => {
                const { component } = setupProps({
                    direction: PlayerDirection.PLAYER_DIRECTION_TOP,
                })
                const instance = component.instance()
                instance.isBubbleActive = () => true
                const receivedClassName = instance.getVoteBubbleStyle()
                expect(receivedClassName).toEqual('bubble bubbleTop active')
            })
        })
    })

    describe('getVoteValue', () => {
        it('returns undefined if vote is undefined or empty string if user has voted but it has not been revealed', () => {
            const { component } = setupProps({
                votes: { mockPlayerName: undefined },
                playerName: 'mockPlayerName',
            })
            const instance = component.instance()
            expect(instance.getVoteValue()).toEqual(undefined)
            component.setProps({ votes: { mockPlayerName: '' } })
            expect(instance.getVoteValue()).toEqual('')
        })

        it('returns JA if user voted for yes', () => {
            const { component } = setupProps({
                votes: { mockPlayerName: true },
                playerName: 'mockPlayerName',
            })
            const instance = component.instance()
            expect(instance.getVoteValue()).toEqual('JA')
        })

        it('returns NEIN if user voted for yes', () => {
            const { component } = setupProps({
                votes: { mockPlayerName: false },
                playerName: 'mockPlayerName',
            })
            const instance = component.instance()
            expect(instance.getVoteValue()).toEqual('NEIN')
        })
    })
})
