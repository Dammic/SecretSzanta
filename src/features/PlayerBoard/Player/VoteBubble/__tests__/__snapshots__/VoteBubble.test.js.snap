// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VoteBubble /> does not render bubble at all if player has not voted (his voting value is undefined) 1`] = `null`;

exports[`<VoteBubble /> renders active top bubble 1`] = `
<div
  className="bubble bubbleTop active"
>
  JA
</div>
`;
